src/Symfony/Component/DependencyInjection/Compiler/ServiceAnnotationsPass.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Symfony\Component\DependencyInjection\Argument\ClosureProxyArgument;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Annotation as Annotations;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\ExpressionLanguage\Expression;
+
+/**
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class ServiceAnnotationsPass implements CompilerPassInterface
+{
+    /**
+     * @var AnnotationReader
+     */
+    private $reader;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!class_exists(AnnotationReader::class)) {
+            return;
+        }
+
+        $this->reader = new AnnotationReader();
```

Comment:
I more than remember since I worked on that again in #21381
Yet it allowed to fix a core issue, which is that there should be no cache enabled at build time.
Thus, using `annotation_reader` could be also seen as a way for us to support using it as a first class citizen - which we have to anyway.